app/controllers/flowcommerce_spree/inventory_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module FlowcommerceSpree
+  class InventoryController < ActionController::Base
+    def stock
+      response = params['items'].inject([]) { |result, item| result << check_stock(item[:id], item[:qty].to_i) }
+      render json: { items: response }, status: :ok
+    end
+
+    private
+
+    def check_stock(flow_number, quantity)
+      variant = Spree::Variant.find_by(sku: flow_number)
+      return { id: flow_number, has_inventory: false } unless variant
+
+      { id: flow_number, has_inventory: variant.available_online?(quantity) }
```

Comment:
`available_online?` means backorderable too, if there are no items in stock. Is this correct?

Reply:
Exactly! That methods checks if the count_on_hand for the Online Stock location is enough to supply the quantity and if not it will check the backorderable value. :) We should be ok